bindings/rust/src/lib.rs

```diff
@@ -893,35 +800,23 @@ mod tests {
         // Successful execution.
         let result = instance.execute("foo", &[]);
         assert!(result.is_ok());
-        let result = result.unwrap();
-        assert!(!result.trapped());
-        assert!(result.value().is_some());
-        assert_eq!(result.value().unwrap().as_u32().unwrap(), 42);
+        assert_eq!(result.unwrap().unwrap().as_u32().unwrap(), 42);
```

Comment:
Idiomatic Rust you say. I think I can see it now...

Reply:
That's only tests. See the example on the top of the file.